### PR TITLE
ISSUES-4006 support rename & modify in single alter

### PR DESCRIPTION
**Pull request by zhang2014**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
#4006 Support rename & modify in single alter

### CODE REVIEW

**Comment by zhang2014** [comment]
Motivation:

in #10851 we will rewrite `ALTER TABLE table_name CHANGE old_column_name new_column_name INT NULL`(MySQL query) to `ALTER TABLE table_name MODIFY COLUMN old_column_name Nullable(Int32), RENAME COLUMN old_column_name TO new_column_name`(ClickHouse query);

**Comment by alesapin** [comment]
As far as I remember there was several problems with simultaneous MODIFY and RENAME. I'll add more tests.

File: tests/queries/0_stateless/01413_modify_and_rename_in_sigle_alter.sql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+DROP TABLE IF EXISTS test_modify_and_rename;
+
+CREATE TABLE test_modify_and_rename(modify_and_rename_column Nullable(String), rename_and_modify_column Nullable(String)) ENGINE = MergeTree() PARTITION BY tuple() ORDER BY tuple();
```
Comment (alesapin) [inline tests/queries/0_stateless/01413_modify_and_rename_in_sigle_alter.sql:3]: Just add some data into this table and probably alter will fail.

Comment (zhang2014) [reply tests/queries/0_stateless/01413_modify_and_rename_in_sigle_alter.sql:3]: Similar query also have this problem, I'll try to fixes it.

Other query for example:
```sql
ALTER TABLE test_modify_and_rename
    RENAME COLUMN modify_and_rename_column TO modify_and_rename_column_new,
    DROP COLUMN modify_and_rename_column_new
```

**Comment by alesapin** [comment]
Hi, @zhang2014. Do you have any plans to continue on this? Maybe we can close it temporarily?

**Comment by zhang2014** [comment]
Ok, let's temporarily close it.

**Comment by zhbdesign** [comment]
> Ok, let's temporarily close it.

Has this bug been solved? Which version can be used

**Comment by zhbdesign** [comment]
> Hi, @zhang2014. Do you have any plans to continue on this? Maybe we can close it temporarily?

Has this problem been solved? Which version is solved

**Pull request closed without merging**